detox/android/detox/src/main/java/com/wix/detox/espresso/action/AdjustSliderToPositionAction.kt

```diff
@@ -0,0 +1,50 @@
+package com.wix.detox.espresso.action
+
+import android.view.View
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.matcher.ViewMatchers
+import com.facebook.react.bridge.JavaOnlyMap
+import com.facebook.react.uimanager.ReactStylesDiffMap
+import com.facebook.react.views.slider.ReactSlider
+import com.facebook.react.views.slider.ReactSliderManager
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers
+
+class AdjustSliderToPositionAction(private val newPosition: Double) : ViewAction {
+    override fun getConstraints(): Matcher<View?>? {
+        return Matchers.allOf(
+            ViewMatchers.isAssignableFrom(
+                ReactSlider::class.java
+            ),
+            getIsDisplayed()
+        )
+    }
+
+    override fun getDescription(): String {
+        return "adjustSliderToPosition"
+    }
+
+    private fun buildStyles(vararg keysAndValues: Any): ReactStylesDiffMap {
+        return ReactStylesDiffMap(JavaOnlyMap.of(*keysAndValues))
+    }
+
+    override fun perform(uiController: UiController?, view: View) {
+        if (view is ReactSlider && newPosition > 0) {
```

Comment:
It's not an error, but technically this would never be called for a non-ReactSlider class due to the constraint

Reply:
I'm doing this in order to perform an explicit cast of view to ReactSlider, since I can't change the function signature